### PR TITLE
[ES-4014] Adding Tableau User and also updating various components

### PR DESCRIPTION
**Pull request by leejona16**
Various changes:
- Adding tableau user/warehouse/monitoring/role
- Adding PDX office and jenkins IPs to network policy
- Tweaking namespace so that everything is upper case
- adding integrations
- adding SSO connection

@jrcjc123 @dan3944

### CODE REVIEW

File: snowflake/warehouse_management/create_integrations.psql [hunk 1]
```diff
@@ -1 +1,73 @@
 -- This script is for the initial creation and management of Snowflake Integrations. Must be run as a prod_data_engineer.
+
+-- file formats
+CREATE OR REPLACE FILE FORMAT qa_skillz_primary.skillz.REDSHIFT_SKILLZ_CSV
```
Comment (dan3944) [inline snowflake/warehouse_management/create_integrations.psql]: Do we expect there to be differences in the file format in qa vs staging vs prod?

File: snowflake/user_management/create_roles.psql [hunk 2]
```diff
@@ -153,135 +227,120 @@ GRANT ALL PRIVILEGES ON WAREHOUSE prod_ap                           TO ROLE prod
...
-GRANT USAGE ON SCHEMA "prod_skillz_primary"."skillz"                TO ROLE prod_server_engineer;
-GRANT SELECT ON ALL TABLES IN SCHEMA "prod_skillz_primary"."skillz" TO ROLE prod_server_engineer;
+GRANT USAGE ON DATABASE prod_skillz_primary                       TO ROLE prod_server_engineer;
+GRANT USAGE ON SCHEMA prod_skillz_primary.skillz                  TO ROLE prod_server_engineer;
+GRANT SELECT ON ALL TABLES IN SCHEMA prod_skillz_primary.skillz   TO ROLE prod_server_engineer;
 
 -- warehouse privileges
+GRANT USAGE ON WAREHOUSE qa_adhoc               TO ROLE qa_server_engineer;
+GRANT USAGE ON WAREHOUSE staging_adhoc          TO ROLE staging_server_engineer;
 GRANT USAGE ON WAREHOUSE prod_engineering_adhoc TO ROLE prod_server_engineer;
```
Comment (dan3944) [inline snowflake/user_management/create_roles.psql]: Why is the naming scheme different for prod than qa/staging (with `engineering`)?

File: snowflake/user_management/create_roles.psql [hunk 3]
```diff
@@ -153,135 +227,120 @@ GRANT ALL PRIVILEGES ON WAREHOUSE prod_ap                           TO ROLE prod
...
+GRANT USAGE ON DATABASE prod_skillz_primary                         TO ROLE prod_ws_engineer;
+GRANT USAGE ON SCHEMA prod_skillz_primary.skillz                    TO ROLE prod_ws_engineer;
+GRANT SELECT ON ALL TABLES IN SCHEMA prod_skillz_primary.skillz     TO ROLE prod_ws_engineer;
+GRANT USAGE ON SCHEMA prod_skillz_primary.corp_mart                 TO ROLE prod_ws_engineer;
+GRANT SELECT ON ALL TABLES IN SCHEMA prod_skillz_primary.corp_mart  TO ROLE prod_ws_engineer;
 
 -- warehouse privileges
+GRANT USAGE ON WAREHOUSE qa_adhoc               TO ROLE qa_ws_engineer;
+GRANT USAGE ON WAREHOUSE staging_adhoc          TO ROLE staging_ws_engineer;
 GRANT USAGE ON WAREHOUSE prod_engineering_adhoc TO ROLE prod_ws_engineer;
```
Comment (dan3944) [inline snowflake/user_management/create_roles.psql]: Same comment as above: why `engineering`?

File: snowflake/warehouse_management/create_integrations.psql (hunk 1 above)
Comment (leejona16) [reply snowflake/warehouse_management/create_integrations.psql]: Generally speaking no, however there is a file format per database so I don't want cross-database dependencies (using the QA file format for a prod database etc)

File: snowflake/user_management/create_roles.psql (hunk 2 above)
Comment (leejona16) [reply snowflake/user_management/create_roles.psql]: We want to split the warehouses between general engineering adhoc queries (data, server, ws etc) and data scientists who potentially will run heavier duty queries. This way we have flexibility to grow them separately and monitor their credit usage separately.

File: snowflake/user_management/create_roles.psql (hunk 3 above)
Comment (leejona16) [reply snowflake/user_management/create_roles.psql]: same as above

**Pull request merged**